### ISSUE TITLE
Template part block doesn't handle a missing template part entity

### ISSUE THREAD

**Issue by talldan** [WordPress/gutenberg#36510]
### Description

When a template part block's associated template part entity/post is missing, the block is designed to show a warning to the user:
https://github.com/WordPress/gutenberg/blob/39f883e503f05aac223fe9d9d03a49d6279192a4/packages/block-library/src/template-part/edit/index.js#L120-L139

Unfortunately this is never shown since the code that determines `isMissing` doesn't work. It uses the value from `getEditedEntityRecord` which always returns a (truthy) object when resolved.

I'm also unsure about this comment next to that code
> A new template part is automatically created if we have any inner blocks but no entity.

This might be outdated as I can't see any evidence that this happens.

To solve this, some design guidance will probably be needed to determine the right behavior.

<strike>Related is #36507, which fixes the same issue in the navigation block. In that change we make the nav block go back to its placeholder when the associated menu id deleted or non-existent.</strike> That regressed and the message is back again 😞 .

### Step-by-step reproduction instructions

1. Add a paragraph with some text in the site editor
2. Make it a template part using the 'Make template part' option in the block settings menu, give it a name, and create it.
3. Save all the changes
4. In wp-admin, navigate to Appearance > Template Parts and delete the template part you just created
5. Go back to the site editor
6. Observe that the template part made in step 2 is now empty. Blocks can be added to it, but they can't be saved.

### Environment info

WordPress trunk
Gutenberg trunk
Brave on Mac OS

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

No

**Comment by talldan** [comment]
#37222 mentions a similar issue with slightly different results.

**Comment by Mamaduka** [comment]
I am assigning this to myself.

I think resolving `isMissing` is the smallest fix we can ship with 5.9. Then, we can discuss a better alternative for the "missing" state for the next release.

P.S. Adding a block to a missing Template Part also triggers an error:
```
actions.js:300 Uncaught TypeError: Cannot read properties of undefined (reading 'blocks')
    at actions.js:300
    at Array.reduce (<anonymous>)
    at actions.js:299
    at thunk-middleware.js:4
    at index.js:24
    at promise-middleware.js:20
    at Object.dispatch (resolvers-cache-middleware.js:57)
    at index.js:312
    at entity-provider.js:199
    at use-block-sync.js:223
```
